From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: 5kft <5kft@5kft.org>
Date: Sat, 4 Aug 2018 14:06:50 +0000
Subject: [ARCHEOLOGY] update R_PIO GPIO block enable (H3/H5)

> X-Git-Archeology: > recovered message: > Move the previous "R_PIO enable" change to initial board gpio_init (from clock_init)
> X-Git-Archeology: - Revision c3f02be362aa216f2d3ee011916f9a18baf58291: https://github.com/armbian/build/commit/c3f02be362aa216f2d3ee011916f9a18baf58291
> X-Git-Archeology:   Date: Sat, 04 Aug 2018 14:06:50 +0000
> X-Git-Archeology:   From: 5kft <5kft@5kft.org>
> X-Git-Archeology:   Subject: update R_PIO GPIO block enable (H3/H5)
> X-Git-Archeology:
> X-Git-Archeology: - Revision 5e8cbd8e6aaa87ee03d08c761d5e765e81c4c686: https://github.com/armbian/build/commit/5e8cbd8e6aaa87ee03d08c761d5e765e81c4c686
> X-Git-Archeology:   Date: Fri, 17 Apr 2020 23:44:35 +0200
> X-Git-Archeology:   From: Igor Pecovnik <igorpecovnik@users.noreply.github.com>
> X-Git-Archeology:   Subject: Switch sunxi and sunxi64 to u-boot v2020.04 (#1894)
> X-Git-Archeology:
> X-Git-Archeology: - Revision f6a09def35e647a5442e0f92f399485be29f19a9: https://github.com/armbian/build/commit/f6a09def35e647a5442e0f92f399485be29f19a9
> X-Git-Archeology:   Date: Thu, 10 Nov 2022 21:49:36 +0100
> X-Git-Archeology:   From: Igor Pecovnik <igorpecovnik@users.noreply.github.com>
> X-Git-Archeology:   Subject: Moving patches to per board and removing obsolete (#4409)
> X-Git-Archeology:
---
 arch/arm/mach-sunxi/board.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm/mach-sunxi/board.c b/arch/arm/mach-sunxi/board.c
index 9a7673d82d..b57f100a43 100644
--- a/arch/arm/mach-sunxi/board.c
+++ b/arch/arm/mach-sunxi/board.c
@@ -26,6 +26,7 @@
 #include <asm/arch/timer.h>
 #include <asm/arch/tzpc.h>
 #include <asm/arch/mmc.h>
+#include <asm/arch/prcm.h>
 
 #include <linux/compiler.h>
 
@@ -78,6 +79,11 @@ ulong board_get_usable_ram_top(ulong total_size)
 #ifdef CONFIG_SPL_BUILD
 static int gpio_init(void)
 {
+#if defined(CONFIG_MACH_SUNXI_H3_H5)
+	/* enable R_PIO GPIO access */
+	prcm_apb0_enable(PRCM_APB0_GATE_PIO);
+#endif
+
 	__maybe_unused uint val;
 #if CONFIG_CONS_INDEX == 1 && defined(CONFIG_UART0_PORT_F)
 #if defined(CONFIG_MACH_SUN4I) || \
-- 
Armbian

